src/compiler/checker.ts

```diff
@@ -14771,6 +14771,49 @@ namespace ts {
             }
         }
 
+        // Static members may conflict with non-configurable non-writable built-in Function object properties 
+        // see https://github.com/microsoft/typescript/issues/442.
+        function checkClassForStaticPropertyNameConflicts(node: ClassLikeDeclaration) {
+            const message = Diagnostics.Static_property_0_conflicts_with_built_in_property_Function_0_of_constructor_function_1;
+            const className = getSymbolOfNode(node).name;
+            for (const member of node.members) {
+                const isStatic = forEach(member.modifiers, (m: Modifier) => m.kind  === SyntaxKind.StaticKeyword);

```

Comment:
should be `getModifierFlags(node) & ModifierFlags.Static` -- this calculates the modifiers once and then caches them